### PR TITLE
fix(html/codegen): prevent omitting body for title

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Bug fix, HTML spec missing that case, I will send a PR to spec

**BREAKING CHANGE:**

No

**Related issue (if exists):**

No

### CODE REVIEW

**Comment by alexander-akait** [comment]
@kdy1 Rebased too :+1:

File: crates/swc_html_codegen/tests/fixture.rs [hunk 1]
```diff
@@ -321,6 +321,41 @@ impl VisitMut for DropSpan {
...
+    }
+
+    fn visit_mut_element(&mut self, n: &mut Element) {
+        n.visit_mut_children_with(self);
+
+        if &*n.tag_name == "body" {
+            if let Some(Child::Text(text)) = n.children.last_mut() {
+                // Drop value from the last `Text` node because characters after `</body>`
+                // moved to body tag
+                text.value = "".into();
```
Comment (kdy1) [inline crates/swc_html_codegen/tests/fixture.rs:344]: `js_word!("")` would be better

File: crates/swc_html_codegen/tests/fixture.rs [hunk 2]
```diff
@@ -321,6 +321,41 @@ impl VisitMut for DropSpan {
...
+            } else {
+                n.children.push(Child::Text(Text {
+                    span: Default::default(),
+                    value: "".into(),
```
Comment (kdy1) [inline crates/swc_html_codegen/tests/fixture.rs:348]: Same here

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

`js_word!()` is compile-time constant

File: crates/swc_html_codegen/tests/fixture.rs (hunk 2 above)
Comment (alexander-akait) [reply crates/swc_html_codegen/tests/fixture.rs:348]: @kdy1 Rebase problem, `NormalizeTest` no need anymore and should be removed :smile:

**Comment by alexander-akait** [comment]
@kdy1 The problem is fixed, sometimes I am feeling `merge=union` is dangerous, beucase we can miss these things

**Comment by alexander-akait** [comment]
@kdy1 Rebased

**Comment by alexander-akait** [comment]
@kdy1 Rebased

**Review by kdy1 — APPROVED** [review]

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_html_codegen

**Pull request merged**